src/main/java/seedu/address/model/person/Email.java

```diff
@@ -40,8 +41,12 @@
      */
     public Email(String email) {
         requireNonNull(email);
-        checkArgument(isValidEmail(email), MESSAGE_CONSTRAINTS);
-        value = email;
+
+        String emailWithDomain = email.contains("@")
+                ? email : email + NUS_DOMAIN;
```

Comment:
Just have a small question here! Since in the `AddComandParser` class we are calling `Email email = new Email(id.studentID)`, where `id` is parsed by `ParserUtil.parseStudentId`, does it mean the `id.studentID` field we pass into the `Email` constructor will always be in the format of `EXXXXXXX`? 

Or is it running concurrently HAHHA so it won't be parsed first.

Reply:
Yes, that's correct leon! It'll always be `EXXXXXXX`. After checking the entire command and all is well, we'll be able to retrieve the **checked/sanitised** `StudentID` and we can use it for the `Email` constructor. Then in the `Email` class, L45-45, the ones you highlighted will then check against that value.

The ternary check is so that when we load our app, behind the scenes, we would check against the data parsed through the objects i.e. `StudentID`, `Phone`, `Email`, etc. It's more like a workaround.

Hope this clarifies.

I might not have understood your comment leon HAHA. Feel free to clarify if any!

Reply:
Ah okay the loading part clears up the ternary check! Forgot about that sorry 😅 

A minor suggestion if you want to include in a future PR - instead of `.contains("@")" we could use `contains(NUS_DOMAIN)` or even `subString(8).equals(NUS_DOMAIN)`. (Just a small point though, should not affect correctness)

Approved the PR! Let's wait for @tiewweijian to approve and merge.

Reply:
No problem on that. Honestly speaking, hard for anyone to see it unless you really trace method by method.

I will take up the suggestion leon. Yeah, I think that's way better than just doing `"@"`. Thanks for pointing it out!

Thank you for reviewing the PR too!